feat: install eslint and add .eslintrc.json file

ShSato4JPN opened the pull request:
close #13

ShSato4JPN committed:
feat: install eslint and add .eslintrc.json file

close #13

piro0919 commented on .eslintrc.json:
```diff
@@ -0,0 +1,28 @@
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": { "project": ["./tsconfig.json"] },
+  "overrides": [],
+  "ignorePatterns": ["/node_modules/", "/dist/"],
```
[imo]

今回のケースでは最適な設定でない気がしますね。 👀

ShSato4JPN replied:
@piro0919 

```
  "parser": "@typescript-eslint/parser",
  "parserOptions": { "project": ["./tsconfig.json"] },
```
typescript を解析するには parser が必要で、その詳細設定として tsconfig.json をオプションで指定する必要があると認識しております。
勉強不足で申し訳ないのですが、こちらの設定が不要な理由が分かりません🤯
公式ドキュメントの例でも実装されていました。（[ESlint](https://eslint.org/docs/latest/user-guide/configuring/configuration-files)）
```
  "overrides": [],
```
使用していないのであれば記述するべきではありませんでした。
```
  "ignorePatterns": ["/node_modules/", "/dist/"]
```
[ESlint-Ignoring Code](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code) を読んだところ、`.eslintignore` ファイルを作成することで lint 対象外のファイル・フォルダを指定できることが分かりました。
なので、ignorePatterns で記述するのは間違っていますね...。
また、dist フォルダは本プロジェクトで使用するか分からないため、現段階では設定に含めるべきではありませんでした。

ShSato4JPN committed:
fix: reconfigure eslint

- reinstall eslint
- reconfigure eslint
- add scripts on package.json

close #13

ShSato4JPN commented:
@piro0919 

[f30ec4f](https://github.com/ShSato4JPN/world-of-zono/pull/14/commits/f30ec4f0426becb4e19b8120628661541c3af6e6)
指摘箇所を修正いたしました。
お手隙の際にご確認よろしくお願いいたします。

piro0919 reviewed (approved):
LGTM 👍

ShSato4JPN commented:
@piro0919 

Thank you for your approval.
